### ISSUE TITLE
There were error messages be thrown out in clusters.log during running ondiscover 

### ISSUE THREAD

**Issue by hu-weihua** [xcat2/xcat-core#5098]
When running ``ondiscover``, there were  error messages be thrown out.
```
Apr 12 02:17:05 briggs01 xcat[69851]: ERR Not a reference at /opt/xcat/sbin/xcatd line 2511.
Apr 12 02:17:05 briggs01 xcat[69851]: ERR Died at /opt/xcat/sbin/xcatd line 2058.
Apr 12 02:17:05 briggs01 xcat[69851]: xcatd: destiny plugin bug, pid 69851, process description: 'xcatd SSL: nextdestiny for mid05tor12cn05@mid05tor12cn05: destiny instance' with error 'Died at /opt/xcat/sbin/xcatd line 2058.#012'
```

But the expected process completed. All items in the chain list work. Not sure what above error message will affect.

The important steps are:
Before discovery:
```
[root@briggs01]#ssh 172.11.139.5 "hostname"
mid05tor12cn05-bmc-test

[root@briggs01]#makegocons -d  mid05tor12cn05

//predefined nodes
[root@briggs01]# lsdef mid05tor12cn05,mid05tor12cn05-bmc     
Object name: mid05tor12cn05
    bmc=mid05tor12cn05-bmc
    bmcvlantag=11
 chain=runcmd=bmcsetup,ondiscover=makegocons|rspconfig:sshcfg|rspconfig:hostname=*,osimage=rhels7.5-alt-rc2-ppc64le-install-compute
    cons=openbmc
    groups=test
    ip=172.12.139.5
    mgt=openbmc
    netboot=petitboot
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    switch=mid05tor12
    switchport=5
Object name: mid05tor12cn05-bmc
    groups=bmc
    ip=172.11.139.5
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
```
After discovery:
```
[root@briggs01 iso]# lsdef mid05tor12cn05
Object name: mid05tor12cn05
    arch=ppc64le
    bmc=mid05tor12cn05-bmc
    bmcvlantag=11
    chain=runcmd=bmcsetup,ondiscover=makegocons|rspconfig:sshcfg|rspconfig:hostname=*,osimage=rhels7.5-alt-rc2-ppc64le-install-compute
    cons=openbmc
    consoleenabled=1
    cpucount=144
    cputype=POWER9 (raw), altivec supported
    currchain=boot
    currstate=install rhels7.5-alt-rc2-ppc64le-compute
    disksize=sda:932GB,sdb:932GB
    groups=test
    ip=172.12.139.5
    mac=70:e2:84:14:28:0e
    memory=523490MB
    mgt=openbmc
    mtm=8335-GTC
    netboot=petitboot
    nodetype=mp
    os=rhels7.5-alt-rc2
    postbootscripts=otherpkgs
    postscripts=syslog,remoteshell,syncfiles
    profile=compute
    provmethod=rhels7.5-alt-rc2-ppc64le-install-compute
    serial=1318C4A
    status=bmcready
    statustime=04-12-2018 02:16:53
    supportedarchs=ppc64
    switch=mid05tor12
    switchport=5

[root@briggs01]# rcons mid05tor12cn05    // console can be connected without running "makegocons mid05tor12cn05"      

[root@briggs01]#ssh 172.11.139.5 "hostname"
mid05tor12cn05-bmc   
```
The related log from ``/var/log/xcat/cluster.log``
```
Apr 12 02:16:39 mid05tor12cn05 xcat.genesis.doxcat:  Received destiny=runcmd=bmcsetup
Apr 12 02:16:39 mid05tor12cn05 xcat.genesis.doxcat:  The destiny=runcmd, destiny parameters=bmcsetup
Apr 12 02:16:48 briggs01 xcat[69824]: DEBUG xcatd: connection from mid05tor12cn05@mid05tor12cn05
Apr 12 02:16:48 briggs01 xcat[69824]: DEBUG xcatd: open new process : xcatd SSL: getbmcconfig for mid05tor12cn05@mid05tor12cn05
Apr 12 02:16:48 briggs01 xcat[69827]: DEBUG xcatd: dispatch request 'getbmcconfig ' to plugin 'bmcconfig'
Apr 12 02:16:48 briggs01 xcat[69824]: xCAT: Allowing getbmcconfig for mid05tor12cn05 from mid05tor12cn05
Apr 12 02:16:48 briggs01 xcat[69827]: DEBUG xcatd: handle request 'getbmcconfig' by plugin 'bmcconfig''s process_request
Apr 12 02:16:48 briggs01 xcat[69824]: DEBUG xcatd: close connection with mid05tor12cn05@mid05tor12cn05
Apr 12 02:16:48 mid05tor12cn05 xcat.genesis.bmcsetup:  BMC Information obtained from xCAT
Apr 12 02:16:48 mid05tor12cn05 xcat.genesis.bmcsetup:  NUMBMCS=1 ==> BMC IP=172.11.139.5/255.255.0.0, GW=172.11.253.27, VLAN=11
Apr 12 02:16:48 mid05tor12cn05 xcat.genesis.bmcsetup:  IPMIVER=2.0, IPMIMFG=42817, XPROD=16975
Apr 12 02:16:48 mid05tor12cn05 xcat.genesis.bmcsetup:  Auto detecting LAN channel...
Apr 12 02:16:48 mid05tor12cn05 xcat.genesis.bmcsetup:  Detected LAN channel 1
Apr 12 02:16:53 briggs01 xcat[6950]: INFO xcatd: install monitor received a connection request from 172.12.139.5
Apr 12 02:16:53 briggs01 xcat[6950]: INFO xcatd: 172.12.139.5 is matched with node mid05tor12cn05
Apr 12 02:16:53 briggs01 xcat[6950]: INFO xcatd: triggering 'updatenodestat mid05tor12cn05 bmcready'...
Apr 12 02:16:53 mid05tor12cn05 xcat:  ready
Apr 12 02:16:53 mid05tor12cn05 xcat:  done
Apr 12 02:16:53 mid05tor12cn05 xcat.genesis.doxcat:  Running nextdestiny 172.12.253.27:3001...
Apr 12 02:16:53 briggs01 xcat[6950]: DEBUG xcatd: dispatch request 'updatenodestat bmcready' to plugin 'updatenode'
Apr 12 02:16:53 briggs01 xcat[6950]: DEBUG xcatd: handle request 'updatenodestat' by plugin 'updatenode''s preprocess_request
Apr 12 02:16:53 briggs01 xcat[6950]: DEBUG xcatd: handle request 'updatenodestat' by plugin 'updatenode''s process_request
Apr 12 02:16:53 briggs01 xcat[6950]: mid05tor12cn05 status: bmcready statustime: 04-12-2018 02:16:53
Apr 12 02:16:53 briggs01 xcat[6950]: INFO xcatd: finish a connection request for mid05tor12cn05 from 172.12.139.5
Apr 12 02:16:53 briggs01 xcat[69848]: DEBUG xcatd: connection from mid05tor12cn05@mid05tor12cn05
Apr 12 02:16:53 briggs01 xcat[69848]: DEBUG xcatd: open new process : xcatd SSL: nextdestiny for mid05tor12cn05@mid05tor12cn05
Apr 12 02:16:53 briggs01 xcat[69851]: DEBUG xcatd: dispatch request 'nextdestiny ' to plugin 'destiny'
Apr 12 02:16:53 briggs01 xcat[69848]: xCAT: Allowing nextdestiny for mid05tor12cn05 from mid05tor12cn05
Apr 12 02:16:53 briggs01 xcat[69851]: DEBUG xcatd: handle request 'nextdestiny' by plugin 'destiny''s process_request
Apr 12 02:16:53 briggs01 xcat[69851]: DEBUG destiny->process_request: starting nextdestiny...
Apr 12 02:16:53 briggs01 xcat[69865]: DEBUG xcatd: connection from root@localhost
Apr 12 02:16:53 briggs01 xcat[69865]: DEBUG xcatd: open new process : xcatd SSL: makegocons to mid05tor12cn05 for root@localhost
Apr 12 02:16:53 briggs01 xcat[69865]: xCAT: Allowing makegocons to mid05tor12cn05 for root from localhost
Apr 12 02:16:53 briggs01 xcat[69866]: DEBUG xcatd: dispatch request 'makegocons mid05tor12cn05 ' to plugin 'goconserver'
Apr 12 02:16:53 briggs01 xcat[69866]: DEBUG xcatd: handle request 'makegocons' by plugin 'goconserver''s preprocess_request
Apr 12 02:16:53 briggs01 xcat[69866]: DEBUG xcatd: handle request 'makegocons' by plugin 'goconserver''s process_request
Apr 12 02:16:53 briggs01 xcat[69865]: DEBUG xcatd: close connection with root@localhost
Apr 12 02:16:53 briggs01 xcat[69851]: DEBUG run ondiscover command: makegocons mid05tor12cn05
Apr 12 02:16:53 briggs01 xcat[69924]: DEBUG xcatd: connection from root@localhost
Apr 12 02:16:53 briggs01 xcat[69924]: DEBUG xcatd: open new process : xcatd SSL: rspconfig to mid05tor12cn05 for root@localhost
Apr 12 02:16:53 briggs01 xcat[69927]: DEBUG xcatd: dispatch request 'rspconfig mid05tor12cn05 sshcfg' to plugin 'openbmc2'
Apr 12 02:16:53 briggs01 xcat[69927]: DEBUG xcatd: handle request 'rspconfig' by plugin 'openbmc2''s preprocess_request
Apr 12 02:16:53 briggs01 xcat[69927]: DEBUG xcatd: handle request 'rspconfig' by plugin 'openbmc2''s process_request
Apr 12 02:16:54 briggs01 xcat[69930]: DEBUG xcatd: dispatch request 'rspconfig mid05tor12cn05 sshcfg' to plugin 'openbmc'
Apr 12 02:16:54 briggs01 xcat[69924]: xCAT: Allowing rspconfig to mid05tor12cn05 sshcfg for root from localhost
Apr 12 02:16:54 briggs01 xcat[69930]: DEBUG xcatd: handle request 'rspconfig' by plugin 'openbmc''s preprocess_request
Apr 12 02:16:54 briggs01 xcat[69940]: DEBUG xcatd: handle request 'getcons' by plugin 'kvm''s process_request
Apr 12 02:16:57 briggs01 xcat[69935]: DEBUG xcatd: close connection with root@localhost
Apr 12 02:17:02 briggs01 xcat[69924]: DEBUG xcatd: close connection with root@localhost
Apr 12 02:17:02 briggs01 xcat[69851]: DEBUG run ondiscover command: rspconfig mid05tor12cn05 sshcfg
Apr 12 02:17:02 briggs01 xcat[69995]: DEBUG xcatd: connection from root@localhost
Apr 12 02:17:02 briggs01 xcat[69995]: DEBUG xcatd: open new process : xcatd SSL: rspconfig to mid05tor12cn05 for root@localhost
Apr 12 02:17:02 briggs01 xcat[69996]: DEBUG xcatd: dispatch request 'rspconfig mid05tor12cn05 hostname=*' to plugin 'openbmc2'
Apr 12 02:17:02 briggs01 xcat[69996]: DEBUG xcatd: handle request 'rspconfig' by plugin 'openbmc2''s preprocess_request
Apr 12 02:17:02 briggs01 xcat[69996]: DEBUG xcatd: handle request 'rspconfig' by plugin 'openbmc2''s process_request
Apr 12 02:17:02 briggs01 xcat[69995]: xCAT: Allowing rspconfig to mid05tor12cn05 hostname=* for root from localhost
Apr 12 02:17:02 briggs01 xcat[69999]: DEBUG xcatd: dispatch request 'rspconfig mid05tor12cn05 hostname=*' to plugin 'openbmc'
Apr 12 02:17:02 briggs01 xcat[69999]: DEBUG xcatd: handle request 'rspconfig' by plugin 'openbmc''s preprocess_request
Apr 12 02:17:03 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://172.11.139.5/login -d '{"data": ["root", "xxxxxx"]}'
Apr 12 02:17:03 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] login 200 OK
Apr 12 02:17:03 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] set_hostname curl -k -c cjar -b cjar -X PUT -H "Content-Type: application/json" https://172.11.139.5/xyz/openbmc_project/network/config/attr/HostName -d '{"data": "mid05tor12cn05-bmc"}'
Apr 12 02:17:04 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] set_hostname 200 OK
Apr 12 02:17:04 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] login curl -k -c cjar -b cjar -X POST -H "Content-Type: application/json" https://172.11.139.5/login -d '{"data": ["root", "xxxxxx"]}'
Apr 12 02:17:05 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] login 200 OK
Apr 12 02:17:05 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] get_netinfo curl -k -c cjar -b cjar -X GET -H "Content-Type: application/json" https://172.11.139.5/xyz/openbmc_project/network/enumerate
Apr 12 02:17:05 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] get_netinfo 200 OK
Apr 12 02:17:05 briggs01 xcat.agent: DEBUG mid05tor12cn05: [openbmc_debug] get_netinfo Found LinkLocal address 169.254.224.119 for interface /xyz/openbmc_project/network/eth0, Ignoring...
Apr 12 02:17:05 briggs01 xcat[69995]: DEBUG xcatd: close connection with root@localhost
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG run ondiscover command: rspconfig mid05tor12cn05 hostname=*
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG destiny->process_request: getdestiny...
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG xcatd: dispatch request 'nodeset enact' to plugin 'petitboot'
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG xcatd: handle request 'nodeset' by plugin 'petitboot''s preprocess_request
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG petitboot: sharedtftp = 1
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG xcatd: handle request 'nodeset' by plugin 'petitboot''s process_request
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG petitboot: running on briggs01
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG petitboot: nodes are mid05tor12cn05
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG petitboot: nodeset did not distribute to the service node
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG petitboot: issue runbeginpre request
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG xcatd: dispatch request 'runbeginpre enact' to plugin 'prescripts'
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG xcatd: handle request 'runbeginpre' by plugin 'prescripts''s preprocess_request
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG prescripts->preprocess_request: no nodes left to process, we are done
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG petitboot: issue setdestiny request
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG xcatd: dispatch request 'setdestiny enact' to plugin 'destiny'
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG xcatd: handle request 'setdestiny' by plugin 'destiny''s process_request
Apr 12 02:17:05 briggs01 xcat[69851]: DEBUG destiny->process_request: starting setdestiny...
Apr 12 02:17:05 briggs01 xcat[69851]: ERR Not a reference at /opt/xcat/sbin/xcatd line 2511.
Apr 12 02:17:05 briggs01 xcat[69851]: ERR Died at /opt/xcat/sbin/xcatd line 2058.
Apr 12 02:17:05 briggs01 xcat[69851]: xcatd: destiny plugin bug, pid 69851, process description: 'xcatd SSL: nextdestiny for mid05tor12cn05@mid05tor12cn05: destiny instance' with error 'Died at /opt/xcat/sbin/xcatd line 2058.#012'
Apr 12 02:17:05 briggs01 xcat[69848]: DEBUG xcatd: close connection with mid05tor12cn05@mid05tor12cn05
Apr 12 02:17:05 mid05tor12cn05 xcat.genesis.doxcat:  nextdestiny - Complete.
Apr 12 02:17:05 mid05tor12cn05 xcat.genesis.doxcat:  The destiny=ondiscover, destiny parameters=ondiscover
Apr 12 02:17:05 mid05tor12cn05 xcat.genesis.doxcat:  Running nextdestiny (ondiscover) 172.12.253.27:3001...
Apr 12 02:17:05 briggs01 xcat[70030]: DEBUG xcatd: connection from mid05tor12cn05@mid05tor12cn05
Apr 12 02:17:05 briggs01 xcat[70030]: DEBUG xcatd: open new process : xcatd SSL: nextdestiny for mid05tor12cn05@mid05tor12cn05
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG xcatd: dispatch request 'nextdestiny ' to plugin 'destiny'
Apr 12 02:17:05 briggs01 xcat[70030]: xCAT: Allowing nextdestiny for mid05tor12cn05 from mid05tor12cn05
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG xcatd: handle request 'nextdestiny' by plugin 'destiny''s process_request
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG destiny->process_request: starting nextdestiny...
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG destiny->setdestiny: state=osimage, target=rhels7.5-alt-rc2-ppc64le-install-compute, action=
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG destiny->setdestiny: issue mkinstall request
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG xcatd: dispatch request 'mkinstall ' to plugin 'anaconda'
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG xcatd: handle request 'mkinstall' by plugin 'anaconda''s preprocess_request
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG xcatd: handle request 'mkinstall' by plugin 'anaconda''s process_request
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: installroot=/install globaltftpdir=/tftpboot
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: xcatmaster=!myipfn! tftpdir=/tftpboot provmethod=rhels7.5-alt-rc2-ppc64le-install-compute
Apr 12 02:17:05 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: imagename=rhels7.5-alt-rc2-ppc64le-install-compute pkgdir=/install/rhels7.5-alt-rc2/ppc64le pkglistfile=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist tmplfile=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl partfile=
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: copy initrd.img and vmlinuz to /tftpboot/xcat/osimage/rhels7.5-alt-rc2-ppc64le-install-compute
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: kcmdline=quiet inst.repo=http://!myipfn!:80/install/rhels7.5-alt-rc2/ppc64le inst.ks=http://!myipfn!:80/install/autoinst/mid05tor12cn05 BOOTIF=70:e2:84:14:28:0e  ip=dhcp  inst.loglevel=debug inst.syslog=!myipfn! kernal=xcat/osimage/rhels7.5-alt-rc2-ppc64le-install-compute/vmlinuz initrd=xcat/osimage/rhels7.5-alt-rc2-ppc64le-install-compute/initrd.img
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG destiny->process_request: getdestiny...
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: dispatch request 'nodeset enact' to plugin 'petitboot'
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'nodeset' by plugin 'petitboot''s preprocess_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: sharedtftp = 1
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'nodeset' by plugin 'petitboot''s process_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: running on briggs01
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: nodes are mid05tor12cn05
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: nodeset did not distribute to the service node
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: issue runbeginpre request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: dispatch request 'runbeginpre enact' to plugin 'prescripts'
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'runbeginpre' by plugin 'prescripts''s preprocess_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG prescripts->preprocess_request: no nodes left to process, we are done
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: issue setdestiny request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: dispatch request 'setdestiny enact' to plugin 'destiny'
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'setdestiny' by plugin 'destiny''s process_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG destiny->process_request: starting setdestiny...
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG destiny->setdestiny: state=osimage, target=rhels7.5-alt-rc2-ppc64le-install-compute, action=
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG destiny->setdestiny: issue mkinstall request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: dispatch request 'mkinstall ' to plugin 'anaconda'
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'mkinstall' by plugin 'anaconda''s preprocess_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'mkinstall' by plugin 'anaconda''s process_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: installroot=/install globaltftpdir=/tftpboot
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: xcatmaster=!myipfn! tftpdir=/tftpboot provmethod=rhels7.5-alt-rc2-ppc64le-install-compute
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: imagename=rhels7.5-alt-rc2-ppc64le-install-compute pkgdir=/install/rhels7.5-alt-rc2/ppc64le pkglistfile=/opt/xcat/share/xcat/install/rh/compute.rhels7.pkglist tmplfile=/opt/xcat/share/xcat/install/rh/compute.rhels7.tmpl partfile=
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: copy initrd.img and vmlinuz to /tftpboot/xcat/osimage/rhels7.5-alt-rc2-ppc64le-install-compute
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG anaconda->mkinstall: kcmdline=quiet inst.repo=http://!myipfn!:80/install/rhels7.5-alt-rc2/ppc64le inst.ks=http://!myipfn!:80/install/autoinst/mid05tor12cn05 BOOTIF=70:e2:84:14:28:0e  ip=dhcp  inst.loglevel=debug inst.syslog=!myipfn! kernal=xcat/osimage/rhels7.5-alt-rc2-ppc64le-install-compute/vmlinuz initrd=xcat/osimage/rhels7.5-alt-rc2-ppc64le-install-compute/initrd.img
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG destiny->process_request: processing is finished for setdestiny
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: starting to handle configuration...
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: Finish to handle configurations
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: issue makedhcp request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: dispatch request 'makedhcp' to plugin 'dhcp'
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'makedhcp' by plugin 'dhcp''s preprocess_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG dhcp: disjointdhcps=0
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG dhcp: hasHierarchy=0
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG dhcp: nodes are mid05tor12cn05
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'makedhcp' by plugin 'dhcp''s process_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG dhcp: sitelogservers= sitentpservers=<xcatmaster> sitenameservers=172.12.253.27 site_domain=pok.stglabs.ibm.com
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG dhcp: load dhcp config file /etc/dhcp/dhcpd.conf
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG petitboot: issue runendpre request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: dispatch request 'runendpre osimage=rhels7.5-alt-rc2-ppc64le-install-compute' to plugin 'prescripts'
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG xcatd: handle request 'runendpre' by plugin 'prescripts''s preprocess_request
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG prescripts->preprocess_request: no nodes left to process, we are done
Apr 12 02:17:06 briggs01 xcat[70033]: DEBUG destiny->process_request: processing is finished for setdestiny
Apr 12 02:17:06 briggs01 xcat[70030]: DEBUG xcatd: close connection with mid05tor12cn05@mid05tor12cn05
Apr 12 02:17:06 mid05tor12cn05 xcat.genesis.doxcat:  nextdestiny (ondiscover) - Complete.
Apr 12 02:17:07 mid05tor12cn05 xcat.genesis.doxcat:  The destiny=install, destiny parameters=install rhels7.5-alt-rc2-ppc64le-compute
Apr 12 02:17:07 mid05tor12cn05 xcat.genesis.doxcat:  Reboot...
```

**Comment by hu-weihua** [comment]
Related pull request #4903

**Comment by hu-weihua** [comment]
Related feature requirement issue #4294

**Comment by cxhong** [comment]
looks like those error message will not affect anything, but we should get rid of those.
Currently, if chain attribute doesn't have other command defined after ondiscover,  
````
chain=runcmd=bmcsetup,ondiscover=makegocons
````
the genesis and xcatd will get into a loop to keep process the ondiscover command until `nodeset nodename osiamge=xxxx` command issued.

@zet809 , can we add `standby` after ondiscover command for the nextdestiny?

**Comment by zet809** [comment]
Hi, @cxhong, you are right that both `runcmd` and `ondiscover` will run into loop with `nextdestiny`, try fix this in #5194, pls help to review. Thx!
But, for the error msg here, still need to fix.

**Comment by hu-weihua** [comment]
Have been verified, thanks @cxhong
